lib/improver/nowcasting/lightning.py

```diff
@@ -431,12 +431,13 @@ def apply_ice(self, prob_lightning_cube, ice_cube):
                     raise ConstraintMismatchError(err_string.format(threshold))
                 # Linearly reduce impact of ice as fcmins increases to 2H30M.
                 ice_scaling = [0., (prob_max * (1. - (fcmins / 150.)))]
-                cube_slice.data = np.maximum(
-                    rescale(ice_slice.data,
-                            data_range=(0., 1.),
-                            scale_range=ice_scaling,
-                            clip=True),
-                    cube_slice.data)
+                if ice_scaling[1] > 0:
```

Comment:
Should this be `ice_scaling[1] >= 0` (given your comment on forecast periods of exactly 2.5 hours)?

Reply:
No, I don't think so. When the forecast period is exactly 2.5 hours, ice_scaling[1] is zero which causes the rescale function to raise an error. This if-test is acceptable because when ice_scaling[1] is zero or lower, the influence of ice on the lightning array is null (or nil?).

Reply:
Oh OK.  So I was thinking of `ice_scaling <= 0` and this is the exact inverse.  Which makes this OK.